src/main/treeDataProviders/utils/analyzerUtils.ts

```diff
@@ -49,6 +51,38 @@ export class AnalyzerUtils {
         return fileNode;
     }
 
+    /**
+     * Get the exclude patterns for the applicable scan base on the exclude pattern for the extension.
+     * The following actions will be preformed to the pattern:
+     * 1. If validation for the exclude pattern fails no exclude will be returned.
+     * 2. If pattern contains {}, it will be splitted to multiple patterns, one for each option
+     * 3. '/**' will be add at the suffix to convert the pattern to match files and not folders
+     * @param log - if log is passed the method will log the error that occur during conversion
+     * @returns the applicable pattern array
+     */
+    public static getApplicableExcludePattern(log?: LogManager): string[] {
```

Comment:
This function deserves a good unit test :)

Reply:
done